api/registrations/serializers.py

```diff
@@ -249,13 +251,36 @@ def create(self, validated_data):
         registration_choice = validated_data.pop('registration_choice', 'immediate')
         embargo_lifted = validated_data.pop('lift_embargo', None)
         reviewer = is_prereg_admin_not_project_admin(self.context['request'], draft)
+        children = validated_data.pop('children', None)
+        if children:
+            draft_root = draft.branched_from
+
+            # Validate to make sure we aren't registering nodes without registering their parents
+            # and all nodes have the same root.
+
+            # First check that all children are valid
+            child_nodes = Node.objects.filter(guids___id__in=children)
+            if child_nodes.count() != len(children):
+                raise exceptions.ValidationError('Some child nodes could not be found.')
+
+            # Second check that all children have a parent being registered. The exception being
+            #  the root node of the registration.
+            nodes_being_registered = list(child_nodes) + [draft_root]
+            for node in child_nodes:
+                parent_node = node.parent_node
```

Comment:
In general, sophisticated validation logic like this should go in the model (fat models, skinny controllers). Then errors raised by the model would get handled here.